### PR TITLE
introduce route generator

### PR DESCRIPTION
**Pull request by cc-cpo**
see https://resocoder.com/2019/04/27/flutter-routes-navigation-parameters-named-routes-ongenerateroute/

### CODE REVIEW

File: lib/routing.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                    otherUserId: args.otherUserId,
+                  ),
+                ),
+              );
+            },
+          );
+        }
+        return _errorRoute();
+      default:
+        return _errorRoute();
```
Comment (BendingBender) [inline lib/routing.dart:58]: not easier to do it only once after the switch-case?

File: lib/pages/other_detail/other_detail_page.dart [hunk 2]
```diff
@@ -5,14 +5,26 @@ import 'package:stimmungsringeapp/data/detail_pages.dart';
...
+
+  OtherDetailPage({this.otherUserId});
+
+  @override
+  _OtherDetailPageState createState() => _OtherDetailPageState();
+}
+
+class _OtherDetailPageState extends State<OtherDetailPage> {
+  @override
+  void didChangeDependencies() {
```
Comment (BendingBender) [inline lib/pages/other_detail/other_detail_page.dart:22]: could you explain this?

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: lib/routing.dart (hunk 1 above)
Comment (cc-cpo) [reply lib/routing.dart:58]: alternative would be break cascade;
my final trick does not work

File: lib/pages/other_detail/other_detail_page.dart (hunk 2 above)
Comment (cc-cpo) [reply lib/pages/other_detail/other_detail_page.dart:22]: see the reso link - he recommends using a Stateful widget to kick off first event

**Pull request merged**